stl/src/awint.h

```diff
@@ -294,57 +290,57 @@ enum wrapKERNEL32Functions {
+using PFNSETTHREADPOOLTIMER              = VOID(WINAPI*)(PTP_TIMER, PFILETIME, DWORD, DWORD);
+using PFNWAITFORTHREADPOOLTIMERCALLBACKS = VOID(WINAPI*)(PTP_TIMER, BOOL);
+using PFNCLOSETHREADPOOLTIMER            = VOID(WINAPI*)(PTP_TIMER);
+using PFNCREATETHREADPOOLWAIT            = PTP_WAIT(WINAPI*)(PTP_WAIT_CALLBACK, PVOID, PTP_CALLBACK_ENVIRON);
+using PFNSETTHREADPOOLWAIT               = VOID(WINAPI*)(PTP_WAIT, HANDLE, PFILETIME);
+using PFNCLOSETHREADPOOLWAIT             = VOID(WINAPI*)(PTP_WAIT);
+using PFNFLUSHPROCESSWRITEBUFFERS        = VOID(WINAPI*)();
+using PFNFREELIBRARYWHENCALLBACKRETURNS  = VOID(WINAPI*)(PTP_CALLBACK_INSTANCE, HMODULE);
+using PFNGETCURRENTPROCESSORNUMBER       = DWORD(WINAPI*)();
+using PFNCREATESYMBOLICLINKW             = BOOLEAN(WINAPI*)(LPCWSTR, LPCWSTR, DWORD);
+using PFNGETTICKCOUNT64                  = ULONGLONG(WINAPI*)();
+using PFNGETFILEINFORMATIONBYHANDLEEX    = BOOL(WINAPI*)(HANDLE, FILE_INFO_BY_HANDLE_CLASS, LPVOID, DWORD);
+using PFNSETFILEINFORMATIONBYHANDLE      = BOOL(WINAPI*)(HANDLE, FILE_INFO_BY_HANDLE_CLASS, LPVOID, DWORD);
+using PFNGETSYSTEMTIMEPRECISEASFILETIME  = VOID(WINAPI*)(LPFILETIME);
+using PFNINITIALIZECONDITIONVARIABLE     = VOID(WINAPI*)(PCONDITION_VARIABLE);
+using PFNWAKECONDITIONVARIABLE           = VOID(WINAPI*)(PCONDITION_VARIABLE);
+using PFNWAKEALLCONDITIONVARIABLE        = VOID(WINAPI*)(PCONDITION_VARIABLE);
+using PFNSLEEPCONDITIONVARIABLECS        = BOOL(WINAPI*)(PCONDITION_VARIABLE, PCRITICAL_SECTION, DWORD);
+using PFNINITIALIZESRWLOCK               = VOID(WINAPI*)(PSRWLOCK);
+using PFNACQUIRESRWLOCKEXCLUSIVE         = VOID(WINAPI*)(PSRWLOCK);
+using PFNTRYACQUIRESRWLOCKEXCLUSIVE      = BOOLEAN(WINAPI*)(PSRWLOCK);
+using PFNRELEASESRWLOCKEXCLUSIVE         = VOID(WINAPI*)(PSRWLOCK);
+using PFNSLEEPCONDITIONVARIABLESRW       = BOOL(WINAPI*)(PCONDITION_VARIABLE, PSRWLOCK, DWORD, ULONG);
+using PFNCREATETHREADPOOLWORK            = PTP_WORK(WINAPI*)(PTP_WORK_CALLBACK, PVOID, PTP_CALLBACK_ENVIRON);
+using PFNSUBMITTHREADPOOLWORK            = VOID(WINAPI*)(PTP_WORK);
+using PFNCLOSETHREADPOOLWORK             = VOID(WINAPI*)(PTP_WORK);
+using PFNCOMPARESTRINGEX = int(WINAPI*)(LPCWSTR, DWORD, LPCWSTR, int, LPCWSTR, int, LPNLSVERSIONINFO, LPVOID, LPARAM);
+using PFNGETLOCALEINFOEX = int(WINAPI*)(LPCWSTR, LCTYPE, LPWSTR, int);
+using PFNLCMAPSTRINGEX   = int(WINAPI*)(LPCWSTR, DWORD, LPCWSTR, int, LPWSTR, int, LPNLSVERSIONINFO, LPVOID, LPARAM);
 
 // Use this macro for caching a function pointer from a DLL
 #define STOREFUNCTIONPOINTER(instance, functionname) \
-    __KERNEL32##Functions[e##functionname] = (PVOID)(GetProcAddress(instance, #functionname));
+    __KERNEL32Functions[e##functionname] = reinterpret_cast<PVOID>(GetProcAddress(instance, #functionname));
 
 // Use this macro as a cached function pointer from a DLL
 #define DYNAMICGETCACHEDFUNCTION(functiontypedef, functionname, functionpointer) \

```

Comment:
Mega-nitpick: `functiontypedef` really should be `functiontype`.

Reply:
Giga-nitpick: This is a function pointer type, not a function type. :smiley_cat:

I've renamed this and the other macro parameters. Please meow if this needs further change.